src/engineio/packet.rs

```diff
@@ -101,66 +117,77 @@ impl Packet {
             },
         })
     }
+}
 
+impl From<Packet> for Bytes {
     /// Encodes a `Packet` into an `u8` byte stream.
-    #[inline]
-    fn encode_packet(self) -> Bytes {
-        let mut result = BytesMut::with_capacity(self.data.len() + 1);
-        result.put((self.packet_id as u8).to_string().as_bytes());
-        result.put(self.data);
+    fn from(packet: Packet) -> Self {
+        let mut result = BytesMut::with_capacity(packet.data.len() + 1);
+        result.put(String::from(packet.packet_id).as_bytes());
+        if packet.packet_id == PacketId::Base64 {
+            result.extend(encode(packet.data).into_bytes());
+        } else {
+            result.put(packet.data);
+        }
         result.freeze()
     }
+}
 
-    // Observed some strange behavior while doing this with socket.io
-    // packets, works with engine.io packets.
-    /// Encodes a `Packet` with the payload as `base64`.
-    #[allow(dead_code)]
-    #[inline]
-    fn encode_base64(self) -> Bytes {
-        assert_eq!(self.packet_id, PacketId::Message);
+pub struct Payload {
+    packets: Vec<Packet>,
+}
```

Comment:
Does this payload struct clashes with the current `socketio::payload::Payload` struct?

Reply:
Shouldn't, its in a different module.